consumer.go

```diff
@@ -93,21 +92,50 @@ func NewConsumerForRemote(ctx context.Context, remote *bfs.Object, opt *Consumer
 		cfn:    cfn,
 	}
 
-	// run initial sync
-	if _, err := c.sync(true); err != nil {
-		_ = c.Close()
+	return c.run()
+}
+
+// NewIncrementalConsumer starts a new incremental feed consumer.
+func NewIncrementalConsumer(ctx context.Context, bucketURL string, opt *ConsumerOptions, cfn ConsumeFunc) (Consumer, error) {
+	bucket, err := bfs.Connect(ctx, bucketURL)
+	if err != nil {
 		return nil, err
 	}
 
-	// start continuous loop
-	go c.loop()
+	csm, err := NewIncrementalConsumerForBucket(ctx, bucket, opt, cfn)
+	if err != nil {
+		_ = bucket.Close()
+		return nil, err
+	}
+	csm.(*consumer).ownRemote = true
+	return csm, nil
+}
 
-	return c, nil
+// NewIncrementalConsumerForBucket starts a new incremental feed consumer with a bucket.
+func NewIncrementalConsumerForBucket(ctx context.Context, bucket bfs.Bucket, opt *ConsumerOptions, cfn ConsumeFunc) (Consumer, error) {
+	var o ConsumerOptions
+	if opt != nil {
+		o = *opt
+	}
+	o.norm()
+
+	ctx, stop := context.WithCancel(ctx)
+	c := &consumer{
+		remote:      bfs.NewObjectFromBucket(bucket, "manifest.json"),
+		bucket:      bucket,
+		opt:         o,
+		ctx:         ctx,
+		stop:        stop,
+		cfn:         cfn,
+		incremental: true,
```

Comment:
I actually don't think we need that, we can determine if something is incremental based on `c.bucket != nil`